### ISSUE TITLE
Identical files not deduped

### ISSUE THREAD

**Issue by petechap** [markfasheh/duperemove#78]
Tested 'master' branch at bc5aae3b19:

duperemove finds the common extent in identical files, but doesn't dedupe them when the '-d' option is given. Sample session here:

https://gist.github.com/petechap/355014a0c752f3d63ccf

Running 'btrfs-extent-same' on the same files manually will dedupe them OK.

For some reason I don't understand, the kernel is marking every file extent I write as 'shared' shortly after being written to disk. However looking at the physical extents in the above gist will clearly show that duperemove is not doing anything

**Comment by petechap** [comment]
Sorry Mark, I only just saw your comment on the gist.

I'm using your 'master branch at bc5aae3, and "uname -a" gives me:
Linux pan 3.19.0-21-generic #21-Ubuntu SMP Sun Jun 14 18:31:11 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

I've bisected the problem and it was actually introduced in revision bc5aae3. Prior to this, the dedupe iotcl is getting issued correctly in my above example. After that revision, no ioctl is issued at all.

**Comment by petechap** [comment]
I was actually about to issue a pull request of my own when you closed issue #73, and I've just found that my own solution doesn't cause this problem:

https://github.com/petechap/duperemove/commit/5355a8e3fc06ff8835444df1c63be0040f4cc166

Hopefully that's enough of a pointer for you to reproduce (and hopefully fix) this problem. Thanks.

**Comment by markfasheh** [comment]
Oh OK i

**Comment by markfasheh** [comment]
Yeah thanks for that. So I fixed any problems regarding double frees, but what I failed to realize was that clean_deduped() needed to run after we get physical offset information in add_shared_extents(). The fix is to move the code back indeed, but we want to catch this case in the worker function to at least avoid reporting of not-real errors.

I should have a patch in master soon to fix this up, if you don't mind testing that it'd be awesome. The long term solution will require breaking dedupe_extent_list() into multiple parts as that portion of the code has gotten.... unruly.
